add set-host command

nicbus opened the pull request:
This PR adds the `set-host` command to set `opret`/`tapret` host, if needed.

This is intended to be called by a user that has prepared a PSBT with an output ready for the commitment but needing to set it as a host, via CLI.

This comes from (and depends on) [rgb-wallet PR #46](https://github.com/RGB-WG/rgb-wallet/pull/46), which initially did the same in the `pay` command but now only makes `opret` and `tapret` modules public.

dr-orlovsky commented on src/bin/rgb/command.rs:
```diff
@@ -667,6 +680,63 @@ impl Command {
+                            {
+                                if !outp.is_opret_host() {
+                                    psbt_modified = true;
+                                    outp.set_opret_host().expect("set opret host");
```
I usually do 
```suggestion
                                    outp.set_opret_host().ok();
```
in such cases

nicbus replied:
done

dr-orlovsky commented on src/bin/rgb/command.rs:
```diff
@@ -667,6 +680,63 @@ impl Command {
+                            {
+                                if !outp.is_tapret_host() {
+                                    psbt_modified = true;
+                                    outp.set_tapret_host().expect("set tapret host");
```
ibid

dr-orlovsky reviewed (changes requested):
Some style improvements - otherwise LGTM. Why draft?

nicbus commented:
> Some style improvements - otherwise LGTM

great, updated

> Why draft?

This PR was set as draft since it could not work until `rgb-wallet` PR 46 was not merged.

note: I now introduced a commit that updates `rgb-wallet` and `rgb-std` to git/master so the `set-host` command can use the now-public `opret`/`tapret` modules.

nicbus commented:
added a commit to update an outdated github action (actually to re-trigger CI as it failed for an internal error on the previous run)

it still failed, though, with more internal errors

the PR IMO is ready, the only test that failed in the previous run (build/cli) succeeds locally (`cargo build --no-default-features --features cli`)

dr-orlovsky commented on src/bin/rgb/command.rs:
```diff
@@ -667,6 +681,51 @@ impl Command {
+                let mut psbt_modified = false;
+                match method {
+                    CloseMethod::OpretFirst => {
+                        if let Some(outp) = psbt
+                            .unsigned_tx
+                            .output
+                            .iter()
+                            .zip(&mut psbt.outputs)
+                            .find(|(o, outp)| {
+                                o.script_pubkey.is_op_return() && !outp.is_opret_host()
+                            })
+                            .map(|(_, outp)| outp)
+                        {
+                            psbt_modified = true;
+                            outp.set_opret_host().ok();
+                        }
```
This is nit: we can optimize it even further

```suggestion
                        psbt
                            .unsigned_tx
                            .output
                            .iter()
                            .zip(&mut psbt.outputs)
                            .find(|(o, outp)| {
                                o.script_pubkey.is_op_return() && !outp.is_opret_host()
                            })
                            .and_then(|(_, outp)| {
                                psbt_modified = true;
                                outp.set_opret_host()
                            })
```

nicbus replied:
idem

dr-orlovsky commented on Cargo.toml:
```diff
@@ -21,8 +21,8 @@ amplify = "4.0.0"
 strict_types = "1.0.0"
-rgb-std = { version = "0.10.0", features = ["fs"] }
-rgb-wallet = { version = "0.10.0", features = ["fs"] }
+rgb-std = { git = "https://github.com/RGB-WG/rgb-wallet", branch = "master", features = ["fs"] }
+rgb-wallet = { git = "https://github.com/RGB-WG/rgb-wallet", branch = "master", features = ["fs"] }
```
Nit: it is better to add this as a separate lines in `[patch.crates-io]` sections to simplify its removal.

nicbus replied:
yes, much better

nicbus commented on src/bin/rgb/command.rs:
```diff
@@ -667,6 +681,51 @@ impl Command {
+                        {
+                            psbt_modified = true;
+                            outp.set_opret_host().ok();
+                        }
```
I like this even more

(still needed the final `.ok()` as an `Option<>` is expected)

nicbus commented:
> ACK with nits. If you want, I can merge it as is; but if you prefer to address nits, this is also an option

liked the suggestions, updated

dr-orlovsky reviewed (approved):
ACK https://github.com/RGB-WG/rgb/commit/2237967c85ae3fef32ae9aa67afd60e79b732a04

Very cool, thank you!
